### ISSUE TITLE
Having in a unionAll creates a misplaced having clause

### ISSUE THREAD

**Issue by ianmaddox** [knex/knex#1630]
Example code

```
var foo = knex.select('id').from("mytable").having('val', '>', '0');
var bar = foo.clone();
bar.unionAll(foo.clone());
bar.unionAll(foo.clone());
console.log(bar.toString());
```

Output:

```
select `id` from `mytable` 
union all
select `id` from `mytable` having `val` > '0'
union all
select `id` from `mytable` having `val` > '0' having `val` > '0'
```

It appears that the union all from the first subquery is applied at the very end.

Tested on version 0.11.10

**Comment by spruce-bruce** [comment]
I've been looking at this a little bit. Seems related to https://github.com/tgriesser/knex/issues/913#issuecomment-240234262

**Comment by ianmaddox** [comment]
Looking at both now, I concur. Guess that helps narrow the source of the problem down to the union logic!

**Comment by kibertoad** [comment]
Released in 1.0.5
